### PR TITLE
Fix issue #709

### LINKED ISSUE #709: ClusterAutoScalerAddOn fails on upgrade from cdk blueprint 1.7.3 -> 1.8.1 & missing versionMap for kubernetes 1.26
### Describe the bug

When upgrading EKS cluster using `ClusterAutoScalerAddOn` with default settings using helm `https://kubernetes.github.io/autoscaler - 9.25.0` and EKS is deployed with cdk blueprint `1.7.3` and upgrading to `1.8.1` `cdk deploy` fails with following error:

```Received response status [FAILED] from custom resource. Message returned: Error: b'Error: UPGRADE FAILED: resource mapping not found for name: "blueprints-addon-cluster-autoscaler-aws-cluster-autoscaler" namespace: "" from "": no matches for kind "PodDisruptionBudget" in version "policy/v1beta1"\nensure CRDs are installed first\n' Logs: /aws/lambda/eks-cluster-name-awscdkawseksK-Handler886CB40B-SjbvK442tNH7 at invokeUserFunction (/var/task/framework.js:2:6) at processTicksAndRejections (internal/process/task_queues.js:95:5) at async onEvent (/var/task/framework.js:1:369) at async Runtime.handler (/var/task/cfn-response.js:1:1573) (RequestId: 9ebc425d-b7be-42c9-8e47-e911be6c4bfe)```

### Expected Behavior

When upgrading EKS cluster using `ClusterAutoScalerAddOn` with default settings using helm `https://kubernetes.github.io/autoscaler` EKS deployed with cdk blueprint `1.7.3` and upgrading to `1.8.1`  `cdk deploy` to finish successfully. 

### Current Behavior

When upgrading EKS cluster using `ClusterAutoScalerAddOn` with default settings using helm `https://kubernetes.github.io/autoscaler - 9.25.0` and EKS is deployed with cdk blueprint `1.7.3` and upgrading to `1.8.1` `cdk deploy` fails with following error:

```Received response status [FAILED] from custom resource. Message returned: Error: b'Error: UPGRADE FAILED: resource mapping not found for name: "blueprints-addon-cluster-autoscaler-aws-cluster-autoscaler" namespace: "" from "": no matches for kind "PodDisruptionBudget" in version "policy/v1beta1"\nensure CRDs are installed first\n' Logs: /aws/lambda/eks-cluster-name-awscdkawseksK-Handler886CB40B-SjbvK442tNH7 at invokeUserFunction (/var/task/framework.js:2:6) at processTicksAndRejections (internal/process/task_queues.js:95:5) at async onEvent (/var/task/framework.js:1:369) at async Runtime.handler (/var/task/cfn-response.js:1:1573) (RequestId: 9ebc425d-b7be-42c9-8e47-e911be6c4bfe)```

### Reproduction Steps

Try to upgrade EKS cluster using `ClusterAutoScalerAddOn` with default settings using helm `https://kubernetes.github.io/autoscaler - 9.25.0` and EKS is deployed with cdk blueprint `1.7.3` and upgrading to `1.8.1` run `cdk deploy`.

### Possible Solution

Use latest version of helm chart for cluster-autoscaler `https://kubernetes.github.io/autoscaler` - `9.29.0` as default setting.

Workaround to upgrade cdk blueprint to `1.8.1` before helm chart for cluster-autoscaler default is updated to latest version. 

```typescript
// When adding blueprint Addons specify version like:
new blueprints.addons.ClusterAutoScalerAddOn({version: '9.29.0'}),

// and then perform cdk deploy using 1.8.1 on exciting cluster  
```

### Additional Information/Context

_No response_

### CDK CLI Version

2.81.0

### EKS Blueprints Version

1.8.1

### Node.js Version

v18.15.0

### Environment details (OS name and version, etc.)

Mac OS 12.6 Monterey - M1 Max

### Other information

_No response_

### PR DESCRIPTION
**Pull request by bnaydenov**
Fix #709

### CODE REVIEW

File: lib/addons/cluster-autoscaler/index.ts [hunk 1]
```diff
@@ -42,8 +42,9 @@ const defaultProps: ClusterAutoScalerAddOnProps = {
  * Version of the autoscaler, controls the image tag
  */
 const versionMap = new Map([
-    [KubernetesVersion.of("1.25"), "9.25.0"],
-    [KubernetesVersion.V1_25, "9.25.0"],
+    [KubernetesVersion.of("1.26"), "9.29.0"],
+    [KubernetesVersion.V1_26, "9.29.0"],
+    [KubernetesVersion.V1_25, "9.29.0"],
     [KubernetesVersion.V1_23, "9.21.0"],
```
Comment (youngjeong46) [inline lib/addons/cluster-autoscaler/index.ts:48]: There's missing one for V1.24, which should be mapped to "9.25.0"

File: lib/addons/cluster-autoscaler/index.ts [hunk 2]
```diff
@@ -42,8 +42,9 @@ const defaultProps: ClusterAutoScalerAddOnProps = {
  * Version of the autoscaler, controls the image tag
  */
 const versionMap = new Map([
-    [KubernetesVersion.of("1.25"), "9.25.0"],
-    [KubernetesVersion.V1_25, "9.25.0"],
+    [KubernetesVersion.of("1.26"), "9.29.0"],
```
Comment (youngjeong46) [inline lib/addons/cluster-autoscaler/index.ts:45]: Lines 45 and 46 look duplicate.

**Review by youngjeong46 — COMMENTED** [review]
Thank you @bnaydenov. Left a couple of comments.

File: lib/addons/cluster-autoscaler/index.ts (hunk 1 above)
Comment (bnaydenov) [reply lib/addons/cluster-autoscaler/index.ts:48]: mapping for V1.24 is added back

File: lib/addons/cluster-autoscaler/index.ts (hunk 2 above)
Comment (bnaydenov) [reply lib/addons/cluster-autoscaler/index.ts:45]: i fully agree with you they look dublicate , but i tried to keep them as close as posssible before my commit. Check the lines in how they look into 1.8.1 with version for V1.25

Comment (shapirov103) [reply lib/addons/cluster-autoscaler/index.ts:45]: that is fine, as long as the map entry exists for the target version of EKS, it is the correct approach. Version may become different in the future.

**Review by shapirov103 — APPROVED** [review]
LGTM

**Pull request merged**